GVRf/Framework/src/org/gearvrf/GVRPointLight.java

```diff
@@ -0,0 +1,139 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gearvrf;
+
+import org.gearvrf.utility.TextFile;
+
+/**
+ * Illuminates object in the scene with a point light source.
+ * 
+ * The position of the light is the position of the scene object
+ * the light is attached to. Light is emitted in all directions
+ * from that point.
+ *
+ * The intensity of the light diminishes with distance from the
+ * light. Three attenuation factors are provided to specify how
+ * the intensity of the light falls off with distance:
+ * {@code 1 / (attenuation_constant + attenuation_linear * D * attenuation_quadratic * D ** 2)
+ *
+ * Point light uniforms:
+ * {@literal
+ *   world_position        position of light in world coordinates
+ *                         derived from scene object position
+ *   ambient_intensity     intensity of ambient light emitted
+ *   diffuse_intensity     intensity of diffuse light emitted
+ *   specular_intensity    intensity of specular light emitted
+ *   attenuation_constant  constant attenuation factor
+ *   attenuation_linear    linear attenuation factor
+ *   attenuation_quadratic quadratic attenuation factor
+ * }
+ * @see GVRPhongDirectLight
+ * @see GVRSpotLight
+ * @see GVRLightBase
+ */
+public class GVRPointLight extends GVRDirectLight
```

Comment:
Every light gets a position and a direction implicitly from the scene object it is attached to. A point light does not use its direction in the shader implementation because it radiates light in all directions. Similarly, a directional light does not use its position in the shader implementation.
I understand you object to this inheritance so I have changed it to have GVRPointLight inherit directly from GVRLightBase.